torch/distributed/benchmarks/benchmark_ddp_rpc.py

```diff
@@ -0,0 +1,350 @@
+
+
+def _print_header():
+    _print_cont("\n")
+    _print_cont("%10s" % "")
+    for p in [50, 75, 90, 95]:
+        _print_cont("%14s%10s" % ("sec/iter", "ex/sec"))
+    _print_cont("\n")
+
+
+def _print_benchmark(prefix, nelem, measurements):
+    measurements = sorted(measurements)
+    _print_cont("%8s:" % prefix)
+    for p in [50, 75, 90, 95]:
+        v = np.percentile(measurements, p)
+        _print_cont("  p%02d:  %1.3fs  %6d/s" % (p, v, nelem / v))
+    _print_cont("\n")
+
+
+def _print_cont(msg):
+    print(msg, end="", flush=True)
+
+
+def _run_printable(cmd):
+    proc = subprocess.run(shlex.split(cmd), capture_output=True)
+    assert proc.returncode == 0
+
+    buffer = io.BytesIO()
+    torch.save(proc.stdout.decode("utf-8"), buffer)
+    input_tensor = torch.ByteTensor(list(buffer.getvalue()))
+    input_length = torch.IntTensor([input_tensor.size(0)])
+
+    output = []
+    buffer = io.BytesIO(np.asarray(input_tensor).tobytes())
+    output.append(torch.load(buffer))
+    return output
+
+
+def _run_trainer(emb_rref_list, rank):
+    r"""
+   Each trainer runs a forward pass which involves an embedding lookup on the
+   8 parameter servers and running nn.Linear locally. During the backward pass,
+   DDP is responsible for aggregating the gradients for the dense part
+   (nn.Linear) and distributed autograd ensures gradients updates are
+   propagated to the parameter servers.
+   """
+
+    # Setup the model.
+    model = HybridModel(emb_rref_list, rank)
+
+    # Retrieve all model parameters as rrefs for DistributedOptimizer.
+
+    # Retrieve parameters from all embedding tables for the current trainer.
+    model_parameter_rrefs = []
+    for ind, emb_rref in enumerate(emb_rref_list):
+        ps_name = "ps{}".format(ind)
+        model_parameter_rrefs.extend(
+            rpc.rpc_sync(ps_name, _retrieve_embedding_parameters, args=(emb_rref,))
+        )
+
+    # model.parameters() only includes local parameters.
+    for param in model.parameters():
+        model_parameter_rrefs.append(RRef(param))
+
+    # Setup distributed optimizer
+    opt = DistributedOptimizer(optim.SGD, model_parameter_rrefs, lr=0.05)
+
+    criterion = torch.nn.CrossEntropyLoss()
+
+    def get_next_batch(rank):
+        for _ in range(10):
+            num_indices = random.randint(20, 50)
+            indices = torch.LongTensor(num_indices).random_(0, NUM_EMBEDDINGS)
+
+            # Generate offsets.
+            offsets = []
+            start = 0
+            batch_size = 0
+
+            while start < num_indices:
+                offsets.append(start)
+                start += random.randint(1, 10)
+                batch_size += 1
+
+            offsets_tensor = torch.LongTensor(offsets)
+            target = torch.LongTensor(batch_size).random_(8).cuda(rank)
+
+            yield indices, offsets_tensor, target
+
+    measurements = []
+    # Include warm-up cycles during training
+    for epoch in range(100 + WARMUP_CYCLES):
```

Comment:
How long does this benchmark take to run? I'd suggest maybe 1000 iterations and 100 WARMUP_CYCLES as the default if it doesn't take too long to run.